nf_requirements/v1alpha1/interface_types.go

```diff
@@ -16,21 +16,43 @@ limitations under the License.
 
 package v1alpha1
 
-type AttachmentType string
-
-// AttachmentTypeNone defines an untagged attachement (no VLAN)
-const AttachmentTypeNone AttachmentType = "none"
-
-// AttachmentTypeVLAN defines a tagged/vlan attachement
-const AttachmentTypeVLAN AttachmentType = "vlan"
-
-type CNIType string
-
-// CNITypeSRIOV defines the sriov cni
-const CNITypeSRIOV CNIType = "sriov"
-
-// CNITypeIPVLAN defines the ipvlan cni
-const CNITypeIPVLAN CNIType = "ipvlan"
-
-// CNITypeMACVLAN defines the macvlan cni
-const CNITypeMACVLAN CNIType = "macvlan"
+import (
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// +kubebuilder:object:root=true
```

Comment:
Curious: are we generating actual CRDs/Open API schemas for these?

Reply:
it was there. Should we remove it. We don't apply to the apiserver but we can use it later for some validations if required.